File: LeetCode/Easy/0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length())
            return false;

        vector<int> ans(26, 0);

        for (auto& c : s) {
            ans[c - 'a']++;
        }

        for (auto& c : t) {
            ans[c - 'a']--;
        }

        for (auto& c : ans) {
            if (c != 0) {
                return false;
            }
        }

        return true;
    }
};

File: LeetCode/Easy/0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function (s, t) {
    if (s.length != t.length) return false;
    const hash = new Array(26).fill(0);
    for (let i = 0; i < s.length; i++) {
        hash[s.charCodeAt(i) - 'a'.charCodeAt(0)]++;
        hash[t.charCodeAt(i) - 'a'.charCodeAt(0)]--;
    }
    for (let i = 0; i < 26; i++) {
        if (hash[i] !== 0) return false;
    }
    return true;
};